src/instructions.rs

```diff
@@ -0,0 +1,814 @@
+use crate::{
+    error::VMError,
+    hardware::{Memory, Register, Registers},
+    trap_routines::{TrapCode, get_c, halt, out, puts, puts_p, trap_in},
+    utils::{sign_extend, update_flags},
+};
+use std::io::{stdin, stdout};
+
+const ONE_BIT_MASK: u16 = 0b1;
+const THREE_BIT_MASK: u16 = 0b111;
+const FIVE_BIT_MASK: u16 = 0b11111;
+const SIX_BIT_MASK: u16 = 0b11_1111;
+const EIGHT_BIT_MASK: u16 = 0b1111_1111;
+const NINE_BIT_MASK: u16 = 0b1_1111_1111;
+const ELEVEN_BIT_MASK: u16 = 0b111_1111_1111;
+
+/// Adds to values and stores the result in a register
+///
+/// This instruction can happen in two different ways. One is
+/// by adding two elements that are in a register each, this is called
+/// register mode. The other mode, known as immediate mode, is to add the value of a register
+/// with the one that is embedded in the instruction itself.
```

Comment:
I think this should be rephrased:

- An instruction does not happen, it's executed.
- I would list the modes for clarity.

For example:

```suggestion
/// This instruction can be executed in two different ways:
///
/// - Register mode: Adds two values taken from the registers storage.
/// - Immediate mode: Adds a value taken from the registers storage with one taken from the instruction itself. The second operand of the addition is taken from a range of bits of the executed instruction.
```